src/main/java/org/openpnp/machine/reference/feeder/ReferenceTrayFeeder.java

```diff
@@ -71,27 +72,32 @@ public void feed(Nozzle nozzle)
 			throws Exception {
 		logger.debug("{}.feed({})", getName(), nozzle);
 		int partX, partY;
+		double JavaBugWorkaroundX,JavaBugWorkaroundY,JavaBugWorkaroundC;
 		
 		if (feedCount >= (trayCountX * trayCountY)) {
 		    throw new Exception(String.format("Tray empty on feeder %s.", getName()));
 		}
         
+	JavaBugWorkaroundC=feedCount; JavaBugWorkaroundY=trayCountY; JavaBugWorkaroundX=trayCountX;
         if (trayCountX >= trayCountY) {
             // X major axis.
-            partX = feedCount / trayCountY;
-            partY = feedCount % trayCountY;
+            JavaBugWorkaroundX = JavaBugWorkaroundC / JavaBugWorkaroundY;
+            JavaBugWorkaroundY = JavaBugWorkaroundC % JavaBugWorkaroundY;
         }
         else {
             // Y major axis.
-            partX = feedCount % trayCountX;
-            partY = feedCount / trayCountX;
+            JavaBugWorkaroundY = JavaBugWorkaroundC / JavaBugWorkaroundX;
+            JavaBugWorkaroundX = JavaBugWorkaroundC % JavaBugWorkaroundX;
         }
+	partX=(int)JavaBugWorkaroundX;
```

Comment:
That's essentially the same as the original code. No need for the intermediate variable.